### ISSUE TITLE
Valid Filename Returns Error Upon Processing

### ISSUE THREAD

**Issue by joncameron** [avalonmediasystem/avalon#5016]
# To Reproduce
Upload a valid media file with the name "ወዳጅህ ማር ቢ.   ሆን ጨርስ.  ህ አትላሰ!@#$^^&@$%&.mov". (https://drive.google.com/file/d/1ie9WFjyaKxEFifNT1WOelxspELmG4D6m/view?usp=share_link)

# Expected Behavior
- [ ] Avalon accepts "ወዳጅህ ማር ቢ.   ሆን ጨርስ.  ህ አትላሰ!@#$^^&@$%&.mov" without error.

# Error

Aws::S3::Errors::NotFound in MasterFilesController#create
Aws::S3::Errors::NotFound
Extracted source (around line #207):

      when 's3'
        self.file_location = file.to_s
        self.file_size = FileLocator::S3File.new(file).object.size
      else
        self.file_location = file.to_s
        self.file_size = file_size

Rails.root: /home/app/avalon
Application Trace | Framework Trace | Full Trace
app/models/master_file.rb:207:in `setContent'
app/services/master_file_builder.rb:42:in `block in from_specs'
app/services/master_file_builder.rb:36:in `each'
app/services/master_file_builder.rb:36:in `from_specs'
app/services/master_file_builder.rb:30:in `build'
app/controllers/master_files_controller.rb:219:in `create'
config/initializers/ldp_solr_duration_logger.rb:25:in `process_action'

**Comment by masaball** [comment]
The character at issue in the filename appears to be the @ sign. When testing in ActiveEncode, removing @ allows the file to process properly.

The line at issue on the ActiveEncode side is where the file is sent to [mediainfo to generate metadata about the file](https://github.com/samvera-labs/active_encode/blob/3cac5a2d69e873313a83656562f0c21efb09c394/lib/active_encode/engine_adapters/ffmpeg_adapter.rb#L47). For some reason the filename does not get properly maintained when sent to the subprocess. Depending on where the @ symbol is in the filename, the error output indicates that the filename becomes something like `file:///home/user/active_encode/spec/fixtures/@/home/user/active_encode/spec/fixtures/filename` or that the filename is truncated (the non-ASCII characters get converted to percent encoding in the error readout. I don't think it is part of the issue since the filename goes through without the @ sign but maybe the interaction of the two is what is at issue). I have not been able to figure out why the @ symbol would be duplicating part of the filepath in that way.

All of the methods that I could find to handle characters in subprocess/shell commands have not worked. These include using `\@`, attempting to change the @ sign to percent/html encoding, explicitly declaring the filename as UTF-8 encoded, wrapping the filename in single or double quotes, etc.

I have also tried all the methods of creating a subprocess for Ruby. This includes wrapping the command in backticks, using %x[], using `system`, and the Open3 library. Open3 in particular is pretty complex and there may be a feature I missed with it, but everything I have tried has failed.

I have tested running the command being sent to the subprocess in a normal terminal, as well as the Rails console, and those have worked. Something seems to be happening specifically with how ActiveEncode is sending the filename that is causing this weird behavior.
